include/circt/Dialect/Calyx/Calyx.td

```diff
@@ -67,14 +67,17 @@ class CalyxContainer<string mnemonic, list<Trait> traits = []> :
   let assemblyFormat = "$body attr-dict";
   let regions = (region SizedRegion<1>: $body);
 
-  let extraClassDeclaration = [{
+  /// Because extraClassDeclaration is overloaded in the subclass, 
+  /// commonClassDeclaration is defined here to help define extraClassDeclaration.
+  code commonClassDeclaration = [{
     /// Returns the body of a Calyx container.
     Block *getBodyBlock() {
       Region* region = &getOperation()->getRegion(0);
       assert(region->hasOneBlock() && "The body should have one Block.");
       return &region->front();
     }
   }];
+  let extraClassDeclaration = commonClassDeclaration;
```

Comment:
I don't think a comment is necessary since this pattern is fairly common.

However, i do prefer the style used here:
https://github.com/llvm/circt/blob/main/include/circt/Dialect/HW/HWStructure.td#L54-L57
(i.e. flipped wrt. how you're doing it now).